TYPO3.Neos/Classes/TYPO3/Neos/Controller/Module/Management/WorkspacesController.php

```diff
@@ -395,7 +395,7 @@ public function publishWorkspaceAction(Workspace $workspace)
         if (($targetWorkspace = $workspace->getBaseWorkspace()) === null) {
             $targetWorkspace = $this->workspaceRepository->findOneByName('live');
         }
-        $workspace->publish($targetWorkspace);
+        $this->publishingService->publishNodes($this->publishingService->getUnpublishedNodes($workspace), $targetWorkspace);
```

Comment:
Hey thank for the feedback. Not sure if you read the commit message? https://github.com/neos/neos-development-collection/pull/763/commits/c1f8f725d8a1ded1d0d379af5aaa9b41b68271a6

Anyway sure that would be great, except that the publish method in the workspace model has it's own logic that doesn't use the publishing service which is needed for those signals to be fired.. hence the change. Now I'd rather change it like this to fix the buggy behavior than start rewriting the publish method in the workspace model to use the publishing service since I wouldn't know who's using it and what side-effect it would lead to.

The whole publishing stuff is a bit messy with some logic in the publishing service and some in workspace model. One example is that there are different signals in the workspace models publish method than the publishing services have.